synce_dev.c

```diff
@@ -278,55 +337,71 @@ static void force_all_eecs_detach(struct synce_dev *dev)
 		dev->last_d_state = state;
 		dev->d_state = state;
 	}
-};
+}
 
 static void dev_update_ql(struct synce_dev *dev)
 {
-	if (dev->ops.update_ql(dev)) {
+	if (update_ql(dev)) {
 		pr_err("update QL fail on %s", dev->name);
 	}
 }
 
 static int rx_ql_changed(struct synce_dev *dev)
 {
+	struct synce_clock_source *c;
 	struct synce_port *p;
 	int ret = 0;
 
-	LIST_FOREACH(p, &dev->ports, list) {
-		ret = synce_port_rx_ql_changed(p);
-		if (ret) {
-			break;
+	LIST_FOREACH(c, &dev->clock_sources, list) {
+		if (c->type == PORT ) {
+			p = c->port;
+			ret = synce_port_rx_ql_changed(p);
+			if (ret) {
+				break;
+			}
 		}
 	}
 
 	return ret;
 }
 
-static struct synce_port *find_dev_best_source(struct synce_dev *dev)
+void set_port_ql_from_ext_src(struct synce_dev *dev, struct synce_port *port,
+			      struct synce_ext_src *ext_src)
+{
+	port->ql_forced = ext_src->ql;
+	if (dev->extended_tlv)
+		port->ext_ql_msg_forced.enhancedSsmCode = ext_src->ext_ql;
+}
+
+static struct synce_clock_source *find_dev_best_clock_source(struct synce_dev *dev)
 {
-	struct synce_port *p, *best_p = dev->best_source;
-
-	LIST_FOREACH(p, &dev->ports, list) {
-		if (best_p != p) {
-			if (synce_port_compare_ql(best_p, p) == p) {
-				pr_debug("old best %s replaced by %s on %s",
-					 synce_port_get_name(best_p),
-					 synce_port_get_name(p), dev->name);
-				best_p = p;
+	struct synce_clock_source *c, *best_c = dev->best_source;
+
+	LIST_FOREACH(c, &dev->clock_sources, list) {
+		if (best_c != c) {
+			if (synce_clock_source_compare_ql(best_c, c) == c) {
+				pr_debug("old ext best %s replaced by %s on %s",
+					 synce_clock_source_get_name(best_c),
+					 synce_clock_source_get_name(c), dev->name);
+				best_c = c;
 			}
 		}
 	}
 
-	if (best_p) {
-		if (synce_port_is_rx_dnu(best_p)) {
+	if (best_c && best_c->type == PORT) {
+		if (synce_port_is_rx_dnu(best_c->port))
 			return NULL;
+	} else if (best_c && best_c->type == EXT_SRC)
```

Comment:
while if is covered within {} we should do the same for else if

Reply:
fixed